docs/interplanetary-file-system.md

```diff
@@ -0,0 +1,67 @@
+
+The beginning of IPFS occurs in 2014, when Juan Benet, a software programmer, creates a company called Protocol Labs. In 2015, Benet announces a new technology that is already known worldwide as IPFS. As mentioned above, the central idea was to build a p2p network, in this way its users could store as well as distribute information to other users, in this way the protocol works in a decentralized manner. For the respective storage, the network uses the Distributed Hash Table ([DHT](https://es.wikipedia.org/wiki/Tabla_de_hash_distribuida)) as well as the [BitTorrent](https://es.wikipedia.org/wiki/BitTorrent) protocol. 
+
+
+## How does IPFS work?
+
+The system is created to search for the data by the content that is being requested and not by location, as it is commonly known. It is important to highlight that IPFS uses the DHT, this function is used to create a hash of the content that you want to add to the network, said hash is unique and will never be repeated, it also redirects the searches to the nodes where the information is found. requested. IPFS uses the SHA-256 hash, the same as [Bitcoin](https://www.newscientist.com/definition/bitcoin/), this due to the benefits it offers such as security, easy to use and does not require many resources to generate the hash.
+
+The current protocol performs the following steps to perform a search:
+
+- First take the web address and then query the DNS for the IP of the server location.
+- Already with the IP in hand, the server information query is made and the respective download is started.
+- After downloading, it shows us the information of said server.
+
+This is how the traditional protocol works, **content searches by location**, in which it is necessary to know the address to which you want to query the data. In the case of IPFS it is different, as mentioned above, the protocol does this search by content. This system uses the following steps to perform a search:
+
+- It is not necessary to know the address, only what is needed is searched in the system.
+- The system sends the search that is being carried out to the network of nodes. This information is protected by encryption, hash and digital signature, this in order to protect the information and give access only to those who are authorized.
+- After sending the search, the nodes respond with information on the available versions of what was requested.
+- By choosing one of the results obtained, you can see both its content and its information history.
+
+That is the way the IPFS system works, for this reason it is known as **search by content** since we simply indicate what we need and the network of nodes responds with the option to consult the data that is going to be required.
```

Comment:
i think they use the term "content addressing"

Reply:
Attended.